coderd/notifications/notificationstest/fake_enqueuer.go

```diff
@@ -97,3 +97,16 @@ func (f *FakeEnqueuer) Sent() []*FakeNotification {
 	defer f.mu.Unlock()
 	return append([]*FakeNotification{}, f.sent...)
 }
+
+func (f *FakeEnqueuer) SentWithTemplateID(id uuid.UUID) []*FakeNotification {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	sent := []*FakeNotification{}
+	for _, notif := range f.sent {
+		if notif.TemplateID == id {
+			sent = append(sent, notif)
+		}
+	}
+	return sent
+}
```

Comment:
suggestion: we could alternatively modify the signature of `FakeEnqueuer.Sent()` to accept an arbitrary number of functions to match notifications we want:

```
func (f *FakeEnqueuer) Sent(matchers ...func(*FakeNotification) bool) []*FakeNotification {
  ...
}
```

Reply:
Have now done so with a matcher for template ids